src/test/java/org/dependencytrack/integrations/defectdojo/DefectDojoClientTest.java

```diff
@@ -0,0 +1,118 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.dependencytrack.integrations.defectdojo;
+
+import org.apache.commons.io.input.NullInputStream;
+import org.apache.http.HttpHeaders;
+import org.dependencytrack.util.HttpUtil;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.rules.ExpectedException;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.integration.ClientAndServer;
+import java.net.URL;
+
+import static org.mockserver.integration.ClientAndServer.startClientAndServer;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.model.ParameterBody.params;
+import static org.mockserver.model.Parameter.param;
+
+public class DefectDojoClientTest {
+
+    private static ClientAndServer mockServer;
+
+    @Rule
+    public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void before() {
+        environmentVariables.set("http_proxy", "http://127.0.0.1:1080");
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        mockServer = startClientAndServer(1080);
+    }
+
+    @AfterClass
+    public static void after() {
+        mockServer.stop();
+    }
+
+    @Test
+    public void testUploadFindingsPositiveCase() throws Exception {
+        String token = "db975c97-98b1-4988-8d6a-9c3e044dfff3";
+        String engagementId = "12345";
+        new MockServerClient("localhost", 1080)
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withHeader(HttpHeaders.AUTHORIZATION, "Token " + token)
+                                .withPath("/defectdojo/api/v2/import-scan")
+                                .withBody(
+                                    params(
+                                        param("engagement", engagementId)
+                                    )
+                                )
+                )
+                .respond(
+                        response()
+                                .withStatusCode(201)
+                                .withHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                );
+        DefectDojoUploader uploader = new DefectDojoUploader();
+        DefectDojoClient client = new DefectDojoClient(uploader, new URL("https://localhost/defectdojo"));
+        client.uploadDependencyTrackFindings(token, engagementId, new NullInputStream(0));
```

Comment:
There is no assertion done in this or the following test.  Since there is no response stubbing to verify a value was returned, perhaps you should look to use the Verification API with MockServer to prove that the expected request was made? https://5-2.mock-server.com/mock_server/verification.html